src/viur/core/i18n.py

```diff
@@ -139,71 +295,175 @@ def parse(self, parser):
-        return str(defaultText).format(kwargs)
+    def _translate(
+        self, key: str, default_text: str, hint: str, kwargs: dict[str, t.Any],
+        translations: dict[str, str], caller
+    ) -> str:
+        """Perform the actual translation during render"""
+        lang = kwargs.pop("force_lang", current.language.get())
+        res = str(translations.get(lang, default_text))
+        return translate.substitute_vars(res, **kwargs)
 
 
 def initializeTranslations() -> None:
     """
-        Fetches all translations from the datastore and populates the *systemTranslations* dictionary of this module.
-        Currently, the translate-class will resolve using that dictionary; but as we expect projects to grow and
-        accumulate translations that are no longer/not yet used, we plan to made the translation-class fetch it's
-        translations directly from the datastore, so we don't have to allocate memory for unused translations.
+    Fetches all translations from the datastore and populates the *systemTranslations* dictionary of this module.
+    Currently, the translate-class will resolve using that dictionary; but as we expect projects to grow and
+    accumulate translations that are no longer/not yet used, we plan to made the translation-class fetch it's
+    translations directly from the datastore, so we don't have to allocate memory for unused translations.
     """
-    global systemTranslations
-
-    invertMap = {}
-
-    for srcLang, dstLang in conf.i18n.language_alias_map.items():
-        if dstLang not in invertMap:
-            invertMap[dstLang] = []
-        invertMap[dstLang].append(srcLang)
-
     # Load translations from static languages module into systemTranslations
+    # If they're in the datastore, they will be overwritten below.
     for lang in dir(languages):
         if lang.startswith("__"):
             continue
-
-        for k, v in getattr(languages, lang).items():
-            if k not in systemTranslations:
-                systemTranslations[k] = {}
-
-            systemTranslations[k][lang] = v
-
-            if lang in invertMap:
-                for i in invertMap[lang]:
-                    systemTranslations[k][i] = v
+        for tr_key, tr_value in getattr(languages, lang).items():
+            systemTranslations.setdefault(tr_key, {})[lang] = tr_value
 
     # Load translations from datastore into systemTranslations
-    for tr in db.Query("viur-translations").run(9999):
-        trDict = {}
-        for lang, translation in tr["translations"].items():
-            trDict[lang] = translation
-            if lang in invertMap:
-                for v in invertMap[lang]:
-                    trDict[v] = translation
-
-        systemTranslations[tr["key"]] = trDict
+    # TODO: iter() would be more memory efficient, but unfortunately takes much longer than run()
+    # for entity in db.Query(KINDNAME).iter():
+    for entity in db.Query(KINDNAME).run(10_000):
```

Comment:
Hmmm... does it make sense to make this configurable?

Reply:
Actually, the value doesn't really make sense. It could also be sys.max or 999999999 if you like.
The method simply needs a high value to be _"unlimited"_.